cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperatorNonParametrizedTest.java

```diff
@@ -148,4 +148,80 @@ public void testCustomLabelsAndAnnotations(VertxTestContext context) {
                     async.flag();
                 })));
     }
-}
\ No newline at end of file
+
+    public void testCASecretsWithOwnerReference(VertxTestContext context) {
+        OwnerReference ownerReference = new OwnerReferenceBuilder()
+                    .withApiVersion("kafka.strimzi.io/v1beta1")
+                    .withKind("Kafka")
+                    .withName(NAME)
+                    .withUid("random-1234")
+                    .withBlockOwnerDeletion(false)
+                    .withController(false)
+                    .build();
+
+
+        Kafka kafka = new KafkaBuilder()
+                .withNewMetadata()
+                    .withName(NAME)
+                    .withNamespace(NAMESPACE)
+                .endMetadata()
+                .withNewSpec()
+                    .withNewKafka()
+                        .withReplicas(3)
+                        .withNewEphemeralStorage()
+                        .endEphemeralStorage()
+                        .withNewTemplate()
+                            .withNewClusterCaCert()
+                                .withNewMetadata()
+                                    .withOwnerReferences(ownerReference)
+                                .endMetadata()
+                            .endClusterCaCert()
+                        .endTemplate()
```

Comment:
BTW: I didn't noticed this before, but this should IMHO not even compile. There is no owner reference support in the template.